src/url-ref.ts

```diff
@@ -1,20 +1,50 @@
-import { ref, watch, Ref } from "vue";
+import { watch, Ref, shallowRef } from "vue";
 import { RouteLocationNormalized, Router } from "vue-router";
 
 export function useUrlRef(router: Router, route: RouteLocationNormalized) {
-  function urlRef(name: string, defaultValue: string): Ref<string> {
-    const value = ref<string>("" + (route.query[name] ?? defaultValue));
+  function urlRef(name: string, defaultValue: boolean): Ref<boolean>;
+  function urlRef(name: string, defaultValue: string): Ref<string>;
+  function urlRef(name: string, defaultValue: number): Ref<number>;
+  function urlRef<T>(name: string, defaultValue: T): Ref<T> {
+    // Parse the value in the URL
+    const type = typeof defaultValue;
+    let startingValue = undefined;
+    const routeParamVal = route.query[name];
+    if (type === "string") {
+      if(routeParamVal !== undefined) {
+        startingValue = routeParamVal + "";
+      }
+    } else if (type === "boolean") {
+      const maybeBoolean = routeParamVal + "";
+      if (maybeBoolean === "1" || maybeBoolean === "true") {
+        startingValue = true;
+      } else if (maybeBoolean === "0" || maybeBoolean === "false") {
+        startingValue = false;
+      }
+    }
+    else if (type === "number") {
+      if(routeParamVal !== undefined) {
+        startingValue = Number.parseInt(routeParamVal + "");
```

Comment:
I suppose this is fine, even if `Number.parseInt()` is a bit weird. For example, it will happily parse `Number.parseInt("0xCat")`

Reply:
yeah true, if you have a better idea thats not tooooo time consuming go ahead

Reply:
I suppose this would be good enough:tm:
```js
if(/^[0-9]+$/.test(routeParamVal + "")) {
   Number.parseInt(routeParamVal + "", 10)
}
```